### PR TITLE
adds activity timeout to version 2.x

### PR DESCRIPTION
**Pull request by opheus2**
This PR adds the ability to add a timeout to automatically lock session if no activity after a set duration of time. 

usage 

```php 
//
use lockscreen\FilamentLockscreen\Http\Middleware\LockerTimer;

//

 $panel->middleware([
      //
      LockerTimer::class,
  ])
```

### CODE REVIEW

**Comment by MarJose123** [comment]
Hi @opheus2 Thank you for your PR.

I would like to take some time to understand the possible cause of this since the logic is only using the middleware instead of mouse or keyboard activity.

I'm thinking to have like this implemented in the frontend rather than in the middleware. Probably similar to this library to detect user activity.

https://vueuse.org/core/useIdle/#useidle

**Comment by opheus2** [comment]
I believe a backend approach is preferable for this, as it's more of a security enhancement.

So, we would just need to include the last activity time in a session for each request, which is quite simple.

If there's no activity for the specified period, we will set the session's lock screen to true. This would be detected during the next request, and the user would need to log in again.

It's simple and straightforward but it's a great addition for people using filament as an admin panel for admins only
@MarJose123

**Comment by opheus2** [comment]
BTW you only need to add the middleware if you need to use it. So there is no breaking change

File: src/Http/Middleware/LockerTimer.php [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+
+class LockerTimer
+{
+    /**
+     * @throws ContainerExceptionInterface
+     * @throws NotFoundExceptionInterface
+     */
+    public function handle($request, Closure $next)
+    {
+        if ($request->method() === 'GET' && $request->session()->get('locker_last_activity') && (time() - $request->session()->get('locker_last_activity')) > config('filament-lockscreen.activity_timeout', 0)) {
```
Comment (MarJose123) [inline src/Http/Middleware/LockerTimer.php]: can you change the default value to 30minutes if the `activity_timeout` doesn't exist and also can you check that the request is not already locked

**Review by MarJose123 — APPROVED** [review]
Thanks 🍻

**Pull request merged**